vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -922,12 +1154,17 @@ def start_load_kv(self, metadata: NixlConnectorMetadata):
         We check for these trnxs to complete in each step().
         """
         for req_id, meta in metadata.requests.items():
+            # NOTE: when host xfer buffer is used, only load kv
+            # for requests with do_load_to_device = True.
+            if self.use_host_buffer and not meta.do_load_to_device:
```

Comment:
I think this should be reverted back to
```suggestion
            if not meta.do_remote_prefill:
```

Reply:
`do_remote_prefill` has been set to `False` in the kv_transfer_params before the request was added into `NixlConnectorMetadata` (by [`update_state_after_alloc` in `NixlConnectorScheduler`]((https://github.com/vllm-project/vllm/blob/main/vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py#L267))). So, NixlConnectorWork can not rely on this attribute to determine which request is waiting for remote kv (a.k.a, do_remote_prefill) when `use_host_memory` is true. Also, in a following comment, you mentioned this attribute should be removed from `ReqMeta`.